tools/slang-test/main.cpp

```diff
@@ -1745,12 +1745,88 @@ void runTestsInDirectory(
     }
 }
 
+struct ToolInvoke
+{
+        /// Returns true if it is a tool invoke
+    bool parse(const char*const* argv, int argc)
+    {
+        m_args.Clear();
+
+        m_binDirectory = ".";
+
+        if (argc < 2 || !_isToolName(argv[1]))
+        {
+            return false;
+        }
+        m_toolName = argv[1];
+
+        // Look for parameters that are for the slang-test, and should be skipped
+        int i = 2;
+        while (i < argc)
+        {
+            if (strcmp(argv[i], "-bindir") == 0 && i + 1 < argc)
+            {
+                m_binDirectory = argv[i + 1];
+                i += 2;
+            }
+            // If nothing found, the rest must be parsed to the tool
+            break;
+        }
+
+        m_args.Add(m_toolName.Buffer());
+        m_args.AddRange(argv + i, argc - i);
+        return true;
+    }
+
+    SlangResult invoke(AppContext* appContext, TestContext* testContext)
+    {
+        // Do I want to strip the -bindir directory that may be later
+
+            // We will just parse everything onto the underlying tool
+        auto func = testContext->getInnerMainFunc(m_binDirectory, m_toolName);
+        if (!func)
+        {
+            AppContext::getStdError().print("error: Unable to launch tool '%s'\n", m_toolName.Buffer());
+            return SLANG_FAIL;
+        }
+
+        return func(AppContext::getSingleton(), testContext->getSession(), int(m_args.Count()), m_args.Buffer());
+    }
+
+    String m_binDirectory;
+    String m_toolName;
+    List<const char*> m_args;
+
+private:
+    static bool _isToolName(const char* name)
+    {
+        static const char* toolNames[] =
+        {
+            "slangc",
```

Comment:
This is obviously a good/easy start for now, but if we set a precedent for how `slang-test` parses its command line then we can just assume that the first argument without a `-` must represent the tool name (and anything after it is tool options, as I advocated above), and we won't have to add anything to a list here when/if we add more test tools.

Reply:
@csyonghe 
> Would you perform a rebase followed by a `push -f` to exclude those commits in this PR?
You're right, it would be better to do it with a rebase. I'll aim to make sure I do it that way going forward.